### ISSUE TITLE
Intersect modifies schema in 0.24

### ISSUE THREAD

**Issue by zedeus** [sinclairzx81/typebox#190]
```ts
import { Type } from "@sinclair/typebox";

const ResponseSchema = Type.Object({
	field: Type.Optional(Type.String())
});

const a = Type.Intersect([ResponseSchema]);
console.log(a.required);

const b = Type.Intersect([ResponseSchema]);
console.log(b.required);
```

Output:
```
undefined
[ 'field' ]
```

Expected output (from 0.23.4):
```
undefined
undefined
```

Commenting out this line fixes it:
https://github.com/sinclairzx81/typebox/blob/116bfef2c2c15cd530f4d18bfaa7058fb360ac87/src/typebox.ts#L612

**Comment by sinclairzx81** [comment]
@zedeus Hi thanks for picking this up! There have been a lot of updates in this release, it's great that you spotted this.

Have removed the `delete schema[Modifier]` and published on `0.24.5`. 

Let me know how you go
S

**Comment by zedeus** [comment]
Nice, thanks! Works great now.
